### PR TITLE
Install pip and cryptography on managed nodes for community.crypto module

### PR DESCRIPTION
**Pull request by wadhwa1**
# Description

Installing pip and cryptography on managed nodes via playbooks itself. This is needed for ansible community.crypto module to work.

Fixes # 
Background - https://github.com/confluentinc/cp-ansible/pull/1214

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Created custom docker image wadhwa1/docker-centos7:v2 so as to first reproduce the cp-env issue via molecule tests and make our tests fail.

Verified via cp-ansible-ondemand-job (Test failed as expected) https://jenkins.confluent.io/job/cp-ansible-on-demand/656/testReport/molecule.mtls-customcerts-rhel.molecule/yml/test/

After fix, here are on demand job run- https://jenkins.confluent.io/job/cp-ansible-on-demand/660/

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: molecule/mtls-customcerts-rhel/molecule.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ platforms:
     hostname: zookeeper1.confluent
     groups:
       - zookeeper
-    image: geerlingguy/docker-centos7-ansible
+    image: wadhwa1/docker-centos7:v2
```
Comment (wadhwa1) [inline molecule/mtls-customcerts-rhel/molecule.yml:12]: created custom docker image `wadhwa1/docker-centos7:v2` so as to first reproduce the cp-env issue via molecule tests and make our tests fail. 

Verified via cp-ansible-ondemand-job  https://jenkins.confluent.io/job/cp-ansible-on-demand/656/testReport/molecule.mtls-customcerts-rhel.molecule/yml/test/

TASK [confluent.platform.ssl : Verify Cert Chain] ******************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ImportError: No module named cryptography
fatal: [zookeeper1]: FAILED! => {"changed": false, "msg": "Failed to import the required Python library (cryptography >= 1.5) on zookeeper1.confluent's Python /usr/bin/python. Please read the module documentation and install it in the appropriate location. If the required library is installed, but Ansible is using the wrong Python interpreter, please consult the documentation on ansible_python_interpreter"}

File: roles/common/tasks/debian.yml [hunk 2]
```diff
@@ -212,3 +212,20 @@
...
       - curl
   tags: package
+
+- name: Install pip
+  apt:
+    name:
+      - python-pip
+    state: present
+  become: true
+  when: ansible_python_version == "2.7.5"
```
Comment (wadhwa1) [inline roles/common/tasks/debian.yml:222]: How can I check if python version is python2.x instead of hardcoding ?

**Comment by erikgb** [comment]
https://www.python.org/doc/sunset-python-2/

Why do we still care for Python 2? It is super-risky to use it. Tons of vulns...

**Comment by khemchandanib** [comment]
Nice one

**Comment by wadhwa1** [comment]
> https://www.python.org/doc/sunset-python-2/
> 
> Why do we still care for Python 2? It is super-risky to use it. Tons of vulns...

some of our users still have python2 on their managed nodes, so, we'll need to support them

**Comment by erikgb** [comment]
> > https://www.python.org/doc/sunset-python-2/
> 
> > 
> 
> > Why do we still care for Python 2? It is super-risky to use it. Tons of vulns...
> 
> 
> 
> some of our users still have python2 on their managed nodes, so, we'll need to support them 💭 

Maybe they should upgrade their OS/python instead of Kafka???

**Comment by nsharma-git** [comment]
> > https://www.python.org/doc/sunset-python-2/
> > Why do we still care for Python 2? It is super-risky to use it. Tons of vulns...
> 
> some of our users still have python2 on their managed nodes, so, we'll need to support them 💭

@wadhwa1 @erikgb we have dropped support for Python2 starting from 7.2.0

File: molecule/mtls-customcerts-rhel/molecule.yml (hunk 1 above)
Comment (nsharma-git) [reply molecule/mtls-customcerts-rhel/molecule.yml:12]: I understand the motive, but do you want to submit these changes ? If thats the case, please move the images to our org registry in docker hub.

File: roles/common/tasks/debian.yml [hunk 3]
```diff
@@ -212,3 +212,22 @@
       - unzip
       - curl
   tags: package
+
+- set_fact:
```
Comment (nsharma-git) [inline roles/common/tasks/debian.yml:216]: We should not be worry about python 2 specially. We should be looking for runtime param of `ansible_python_interpreter` It's not just matter of Python 2 vs 3 but there can be multiple versions of python on managed nodes, and user may want to use one of them for many reasons.

**Review by nsharma-git — CHANGES_REQUESTED** [review]

File: molecule/mtls-customcerts-rhel/molecule.yml (hunk 1 above)
Comment (wadhwa1) [reply molecule/mtls-customcerts-rhel/molecule.yml:12]: There's another ticket in sprint board to create custom images, maintain respective dockerfiles, and host images onto our org registry. we can skip submitting these changes for now, I had added them solely for reproducing the scenario and testing changes. I'll revert these/ remove them from this changeset. sounds okay to you ?

File: roles/common/tasks/debian.yml (hunk 3 above)
Comment (wadhwa1) [reply roles/common/tasks/debian.yml:216]: Let’s discuss this, 
As far as I understood, `ansible_python_version` is another runtime param which we have used now to detect the selected python version, 
ansible_python_interpreter is set to auto (default), auto_legacy, auto_legacy_silent, auto_silent
References - 
https://sleeplessbeastie.eu/2020/02/07/how-to-instruct-ansible-to-use-specific-version-of-python/
https://docs.ansible.com/ansible/latest/reference_appendices/interpreter_discovery.html

File: molecule/mtls-customcerts-rhel/molecule.yml (hunk 1 above)
Comment (wadhwa1) [reply molecule/mtls-customcerts-rhel/molecule.yml]: done, reverted this temp change

**Review by nsharma-git — APPROVED** [review]

**Pull request merged**